cmd/snap-gdb-shim/snap-gdbserver-shim.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "../libsnap-confine-private/utils.h"
+
+int main(int argc, char **argv) {
+    if (sc_is_debug_enabled()) {
+        for (int i = 0; i < argc; i++) {
+            printf("-%s-\n", argv[i]);
+        }
+    }
+    // Signal to "snap run" that we are ready to get a debugger attached.
+    // When a debugger attached it will stop the binary at whatever
+    // point the binary is executing. So we cannot have clever code
+    // here that e.g. waits for a debugger to get attached because that
+    // code would also get stoppped/debugged by that debugger and that
+    // would be confusing for the user.
+    //
+    // once a debugger is attached we expect it to send:
+    //  "continue; signal SIGCONT"
+    raise(SIGSTOP);
+
+    // XXX: should we stop here so that users can do stuff before
```

Comment:
Is this comment still valid? AFAIU the welcome message & the spread test, when the user connects with gdb, he can decide when to run the app?

As for the question IMHO "not running" should be the default behavior, this is what you normally get with `gdb foo` and this is the moment one usually wants to set up brakepoints, tracepoints or attach macros.